### PR TITLE
Add maxsplit argument to strutils.split

### PR DESCRIPTION
**Pull request by mjoud**
This implements a `maxsplit` argument to `strutils.split` that limits the number of splits to at most `maxsplits`, yielding `maxsplit+1` substrings. This feature is something I've had the need for when working with csv files and similar, when only the first couple of fields are of interest. The behaviour should be (if implemented correctly!) be identical to that of `str.split` in Python.

I'm not sure if the tests are sufficient - there were no tests for `split` to start with.

### CODE REVIEW

File: lib/pure/strutils.nim [hunk 1]
```diff
@@ -367,15 +367,20 @@ iterator split*(s: string, seps: set[char] = Whitespace): string =
...
   var last = 0
+  var splits = maxsplit
   assert(not ('\0' in seps))
   while last < len(s):
     while s[last] in seps: inc(last)
     var first = last
     while last < len(s) and s[last] notin seps: inc(last) # BUGFIX!
     if first <= last-1:
+      if splits == 0:
+        yield substr(s, first, len(s)-1)
```
Comment (yglukhov) [inline lib/pure/strutils.nim]: AFAIK, every yield inside an inline iterator causes a copy of the `for` code and thus extra yields have to be avoided.

Comment (mjoud) [inline lib/pure/strutils.nim]: Hmm, didn't see that one coming. Any idea how to overcome this?

Comment (yglukhov) [inline lib/pure/strutils.nim]:
```nim
if splits == 0: last = len(s)
yield substr(s, first, last-1)
if splits == 0: break
```

**Comment by mjoud** [comment]
Updated according to comment.

**Comment by Araq** [comment]
I've voted against such a PR in the past:

* parsing CSV with split is like parsing HTML with regexes. It doesn't really work. We have a parsecsv module in the stdlib.
* I still cannot think of other good use cases.

But since it comes up regularly, maybe I'm wrong.

**Comment by kirbyfan64** [comment]
I actually use this a *lot* with parsing restricted DSLs in Python.

**Comment by mjoud** [comment]
I think this method is expected for people coming from other languages, and I'd say it's one of the Python methods I use the most (hence this PR). I'd be happy to close, though, if enough you vote against.

**Comment by mjoud** [comment]
Found #662

**Comment by dom96** [comment]
Why is travis failing?

**Comment by rbehrends** [comment]
There are more general solutions available that don't require hacking `split()`, for example:

```nim
iterator first*(iter: expr, n: Natural): auto =
  var i = 0
  for x in iter:
    if i >= n:
      break
    i += 1
    yield x

import strutils, sequtils
echo toSeq(first(split("foo bar baz"), 2))
```

**Comment by dom96** [comment]
Maybe a ``first`` proc would indeed be better? Should the ``n`` arg be first maybe? ``first 5`` reads better.

**Comment by Araq** [comment]
Merged manually.

**Pull request merged**